src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -471,27 +506,64 @@ class Simplex_tree {
     auto root_source = complex_source.root_;
 
     // root members copy
-    root_.members() = Dictionary(boost::container::ordered_unique_range, root_source.members().begin(), root_source.members().end());
+    root_.members() =
+        Dictionary(boost::container::ordered_unique_range, root_source.members().begin(), root_source.members().end());
     // Needs to reassign children
     for (auto& map_el : root_.members()) {
       map_el.second.assign_children(&root_);
     }
-    rec_copy(&root_, &root_source);
+    rec_copy<Options::store_key>(
+        &root_, &root_source, [](const Filtration_value& fil) -> const Filtration_value& { return fil; });
+  }
+
+  // Copy from complex_source to "this"
+  template<typename OtherSimplexTreeOptions, typename F>
+  void copy_from(const Simplex_tree<OtherSimplexTreeOptions>& complex_source, F&& translate_filtration_value) {
+    null_vertex_ = complex_source.null_vertex_;
+    filtration_vect_.clear();
+    dimension_ = complex_source.dimension_;
+    auto root_source = complex_source.root_;
+
+    // root members copy
+    if constexpr (!Options::stable_simplex_handles) root_.members().reserve(root_source.size());
+    for (auto& p : root_source.members()){
+      if constexpr (Options::store_key && OtherSimplexTreeOptions::store_key) {
+        auto it = root_.members().try_emplace(
+            root_.members().end(),
+            p.first,
+            Node(&root_, translate_filtration_value(p.second.filtration()), p.second.key()));
```

Comment:
Not necessary for this PR (the next one affects of lot of `emplace` calls), but note that an advantage of `try_emplace` is that it builds the mapped type automatically from the arguments you pass, you don't need to construct it yourself.
```suggestion
        auto it = root_.members().try_emplace(
            root_.members().end(),
            p.first,
            &root_, translate_filtration_value(p.second.filtration()), p.second.key());
```